spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminTemplate.java

```diff
@@ -107,11 +110,27 @@ public SchemaFactory getSchemaFactory() {
 	@Override
 	public void createTable(boolean ifNotExists, CqlIdentifier tableName, Class<?> entityClass,
 			Map<String, Object> optionsByName) {
+		this.createTable(ifNotExists, entityClass, optionsByName);
+	}
 
+	@Override
+	public void createTable(boolean ifNotExists, Class<?> entityClass, Map<String, Object> optionsByName) {
 		CassandraPersistentEntity<?> entity = getConverter().getMappingContext().getRequiredPersistentEntity(entityClass);
 
 		CreateTableSpecification createTableSpecification = this.schemaFactory
-				.getCreateTableSpecificationFor(entity, tableName).ifNotExists(ifNotExists);
+				.getCreateTableSpecificationFor(entity, entity.getTableName())
+				.ifNotExists(ifNotExists);
+
+		if (!CollectionUtils.isEmpty(optionsByName)) {
```

Comment:
This is actually a bugfix. Can you please split this pull request into two PR's, one for introducing an overload without accepting a table name and the other one to consider options?